src/sinks/kafka/request_builder.rs

```diff
@@ -1,42 +1,43 @@
-use std::num::NonZeroUsize;
-
-use bytes::{Bytes, BytesMut};
+use bytes::Bytes;
+use lookup::OwnedTargetPath;
 use rdkafka::message::{Header, OwnedHeaders};
-use tokio_util::codec::Encoder as _;
-use vrl::path::OwnedTargetPath;
 
 use crate::{
-    codecs::{Encoder, Transformer},
-    event::{Event, Finalizable, Value},
-    internal_events::{KafkaHeaderExtractionError, TemplateRenderingError},
+    internal_events::KafkaHeaderExtractionError,
     sinks::{
         kafka::service::{KafkaRequest, KafkaRequestMetadata},
-        util::metadata::RequestMetadataBuilder,
+        prelude::*,
     },
-    template::Template,
 };
 
 pub struct KafkaRequestBuilder {
     pub key_field: Option<OwnedTargetPath>,
     pub headers_key: Option<OwnedTargetPath>,
-    pub topic_template: Template,
-    pub transformer: Transformer,
-    pub encoder: Encoder<()>,
+    pub encoder: (Transformer, Encoder<()>),
 }
```

Comment:
Modified `KafkaRequestBuilder` to implement `RequestBuilder` rather than being a one-off implementation.